registry/client/repository.go

```diff
@@ -114,9 +114,7 @@ func (r *registry) Repositories(ctx context.Context, entries []string, last stri
 			return 0, err
 		}
 
-		for cnt := range ctlg.Repositories {
-			entries[cnt] = ctlg.Repositories[cnt]
-		}
+		copy(entries, ctlg.Repositories)
```

Comment:
This looks unrelated as well; looks like it was part of https://github.com/distribution/distribution/commit/fbdfd1ac3580f1aafdbd17c09ebd02c86ccfc4a1; which is 

- https://github.com/distribution/distribution/pull/3711

Not sure if we want to backport that as a whole (it seems rather large), but if we want, should be separate.

Reply:
copy(entries, ctlg.Repositories) is for goci-lint failure.

Reply:
It's not easy to decouple the fix for golangci-lint from the version upgrade of go, because there is some correlation between the versions of go and golangci-lint.

So, in general, you would want us to fix all the prerequisites for the go upgrade, and then have this pr focus only on the go version upgrade.